### PR TITLE
mkdocs: 1.0.4 -> 1.2.1

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.2.1

Change log: https://www.mkdocs.org/about/release-notes/

```bash
:~/nixpkgs]$ ./result/bin/mkdocs new tests
INFO     -  Creating project directory: tests
INFO     -  Writing config file: tests/mkdocs.yml
INFO     -  Writing initial docs: tests/docs/index.md
```

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 127314` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>mkdocs</li>
    <li>python38Packages.pyyaml-env-tag</li>
    <li>python39Packages.pyyaml-env-tag</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127314` at 08bee599 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>mkdocs</li>
    <li>python38Packages.pyyaml-env-tag</li>
    <li>python39Packages.pyyaml-env-tag</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyyaml-env-tag/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, pyyaml
+, pytestCheckHook
+}:
+
+buildPythonPackage
+rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyyaml-env-tag/default.nix:10]:
```suggestion
buildPythonPackage rec {
```

Comment (fabaff) [reply pkgs/development/python-modules/pyyaml-env-tag/default.nix]: Thanks, line break remove.

**Pull request merged**